### PR TITLE
RI-3892 - Tree View Improvements

### PR DESCRIPTION
**Pull request by nmammadli**

### CODE REVIEW

File: tests/e2e/pageObjects/browser-page.ts [hunk 1]
```diff
@@ -1013,6 +1014,39 @@ export class BrowserPage {
             .click(this.selectIndexDdn)
             .click(option);
     }
+
+    /**
+    * Get text from first tree element
+    */
+
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/browser-page.ts:1021]: Redundant space

File: tests/e2e/pageObjects/browser-page.ts [hunk 2]
```diff
@@ -1013,6 +1014,39 @@ export class BrowserPage {
...
+    async getTextFromFirstTreeElement(): Promise<String> {
+        return (await Selector(`[role="treeitem"]`).nth(0).find(`div`).textContent).replace(/\s/g, '');
+    }
+
+    /**
+    * Get text from first tree element
+    * @param names folder names with sequence of subfolder
+    * Example: if names ['mobile', '2']
+    * It will go to mobile -> 2 -> keys
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/browser-page.ts:1030]: IMO it's not required. We generally do that where we have "complexity."
@vlad-dargel - WDYT?

File: tests/e2e/pageObjects/cli-page.ts [hunk 3]
```diff
@@ -171,4 +171,19 @@ export class CliPage {
         await t.click(this.readMoreButton);
         await t.expect(getPageUrl()).eql(url, 'The opened page not correct');
     }
+
+    /**
+    *  Create random index name with CLI and return
+    */
+
+    async createRandomIndexNamewithCLI(): Promise<string> {
+        let word = common.generateWord(10);
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/cli-page.ts:180]: Could be `const`

File: tests/e2e/pageObjects/cli-page.ts [hunk 4]
```diff
@@ -171,4 +171,19 @@ export class CliPage {
...
+        let index = 'idx:' + word;
```
Comment (ofersteinberg) [inline tests/e2e/pageObjects/cli-page.ts:181]: 1. Could be `const`
2. Better method: 
```
 `idx: ${word}`
```

File: tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e copy.ts [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+
+const cliPage = new CliPage();
+
+let keyNames: string[];
+
+let keyName1: string;
+let keyName2: string;
+let keynameSingle: string;
+
+let index: string;
```
Comment (ofersteinberg) [inline tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e copy.ts:27]: Please remove the redundant spaces

File: tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e copy.ts [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+
+        await t.click(browserPage.treeViewButton);
+        await browserPage.openTreeFolders(folders);
+        await t.click(browserPage.refreshKeysButton);
+
+        await t.expect(
+            Selector(`[data-testid="node-item_${folders[0]}:${folders[1]}:keys:keys:"]`).visible)
+            .ok("Folder is not selected");
+
+    });
```
Comment (vlad-dargel) [inline tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e copy.ts:140]: Don't see these tests:
1) "The folder without any patterns selected and the list of keys is displayed when there is a folder without any patterns"
image
2) The folder without any namespaces is selected (if exists) when folder does not exist after search/filter
image
3) The first folder with namespaces is expanded and selected when folder and folder without any namespaces does not exist after search/filter
(it is when one folder is opened and you search for key from other folder - other folder should be opened after such search)
4) The previously selected folder is preselected when key does not exist after keys refresh (you need to select one folder, then delete any key from this folder by CLI and click refresh - verify that the same folder still selected)

**Review by vlad-dargel — CHANGES_REQUESTED** [review]

File: tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts [hunk 7]
```diff
@@ -0,0 +1,169 @@
...
+    .page(commonUrl);
+test
+    .before(async () => {
+        await acceptLicenseTermsAndAddDatabaseApi(ossStandaloneConfig, ossStandaloneConfig.databaseName);
+    })
+    .after(async () => {
+        await t.click(browserPage.patternModeBtn);
+        await browserPage.deleteKeysByNames(keyNames);
+        await deleteStandaloneDatabaseApi(ossStandaloneConfig);
+    })('Tree view preselected folder, Filterred Tree view preselected folder, Refreshed Tree view preselected folder, Search capability Filterred Tree view preselected folder', async t => {
```
Comment (vlad-dargel) [inline tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts:36]: too long name
you can name it just 'Tree view preselected folder'

and below before each ".expect/verify" in test add comment with test from checklist (to be understandable what each verification step actually checks)

File: tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts [hunk 8]
```diff
@@ -0,0 +1,169 @@
...
+
+test
+    .before(async () => {
+        await acceptLicenseTermsAndAddDatabaseApi(ossStandaloneConfig, ossStandaloneConfig.databaseName);
+    })
+    .after(async () => {
+        await t.click(browserPage.patternModeBtn);
+        await browserPage.deleteKeysByNames(keyNames.slice(1));
+        await deleteStandaloneDatabaseApi(ossStandaloneConfig);
+    })('Refreshed Tree view preselected folder, Filterred Tree view preselected folder', async t => {
```
Comment (vlad-dargel) [inline tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts:143]: the name is the same as part of the 1st test name from this fixture

**Review by vlad-dargel — COMMENTED** [review]
Looks good
just need to map all verifications (.expect and verifyKeysDisplayedInTheList) with tests from checklist.
you can use names both from Scenario and Checklist column
image

File: tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts [hunk 9]
```diff
@@ -0,0 +1,169 @@
...
+        await t.click(browserPage.clearFilterButton);
+        await t.expect(firstTreeItemKeys.visible).ok('Folder is not selected');
+        await verifyKeysDisplayedInTheList([`${firstTreeItemText}:1`, `${firstTreeItemText}:2`]); // verify created keys are visible
+
+        await browserPage.selectFilterGroupType(KeyTypesTexts.Stream);
+        await t.expect(browserPage.keyListTable.textContent).contains('No results found.', 'Key is not found message not displayed');
+
+        await t.click(browserPage.streamDeleteButton); // clear stream from filter
+        await t.expect(
+            Selector('[role="rowgroup"]').find('div').withText('No results found.').visible)
```
Comment (vlad-dargel) [inline tests/e2e/tests/critical-path/tree-view/tree-view-improvements.e2e.ts:106]: just use here
`await t.expect(browserPage.keyListTable.textContent).notContains('No results found.', 'Key is not found message still displayed');`

**Review by vlad-dargel — APPROVED** [review]

**Pull request merged**